Update `<OptionRow />` disabled styles

t-wright opened the pull request:
## Description

Currently when disabled, the `<OptionRow />` component retains some styling which may make it look like it's still enabled. Namely, the background colour change on hover, and the arrow.  This PR removes these styles if the component is disabled. In 91785be65964540f8b33139e78fbd729d410a0e9 I explored further changes that could potentially be applied in a disabled state, but decided that these were maybe too opinionated. 

#### Before
![Screen Capture on 2022-06-02 at 13-09-02](https://user-images.githubusercontent.com/4177859/171544959-0082b81c-a00c-4bfa-98fb-b8a1026f036f.gif)

#### After
![Screen Capture on 2022-06-02 at 13-09-56](https://user-images.githubusercontent.com/4177859/171544954-aaa14117-3bb0-4d52-a4d7-fad8b79f0d5c.gif)

## Checklist

- [ ] This PR has good automated test coverage
- [x] The storybook for the component has been updated

## Steps to Test

1. Pull down PR.
2. `npm run build`
1. Copy the `vip-design-system/build` contents into the `vip-ui/node_modules/@automattic/vip-design-system/build` folder
1. Start the vip-ui `npm run dev`
1. Go to http://localhost:3000/apps/2003/production/data
1. View the disabled `<OptionRow />` at at the top of the page and verify hover style and arrow are not displayed.

t-wright committed:
Add to `<OptionRow />` disabled styles

The `disabled` prop currently doesn’t fully disable the component. This change makes it so that when it’s disabled, it renders as a div without a title or href, while hiding the arrow and removing the background on hover.

t-wright committed:
Don’t overwrite `as` , `to`, or `title` props

Rollback these changes introduced in the previous commit. They are probably heavy-handed and could reduce flexibility of the component.

t-wright committed:
Update Storybook component

mjangda commented:
Can you add a before + after screenshot or video?

brunobasto commented on src/system/OptionRow/OptionRow.js:
```diff
@@ -107,7 +107,7 @@ const OptionRow = ( {
 				{body && <Text sx={{ mb: 0 }}>{body}</Text>}
 			</Box>
-			<Box>{meta ? meta : <MdArrowForward size={24} />}</Box>
+			<Box>{meta ? meta : ! disabled && <MdArrowForward size={24} />}</Box>
```
I wonder if we should not couple the `disabled` prop to the `meta` prop this way. It may be valuable to display the arrow even when the row is disabled - to indicate that if that option was enabled, it would take you somewhere, for example.

t-wright replied:
That's a good point. Perhaps it would be better to just allow removing the arrow separately? E.g. by explicitly meta to `false` – `<OptionRow meta={false}  />`? What do you think @brunobasto ?

```suggestion
			{false !== meta && (
				<Box>{meta ? meta : <MdArrowForward size={24} />}</Box>
			)}
```

t-wright replied:
Done in [0b9ae26](https://github.com/Automattic/vip-design-system/pull/63/commits/0b9ae2644b0110f5827261d9aa00a89f3e51698f)

brunobasto reviewed (approved):
LGTM. Left a minor comment. Thank you!

t-wright committed:
Don’t remove meta if `disabled`

Rather than hiding meta if the component is disabled, allow it to be hidden by setting meta to `false` or an empty string.
